### PR TITLE
Convert signup page to typescript.

### PR DESCRIPTION
**Pull request by oponder**
Fixes the issue with not being signed in after creating your account too.

### CODE REVIEW

File: src/components/SignUp/SignUp.tsx [hunk 1]
```diff
@@ -171,14 +206,15 @@ export class SignUp extends React.Component {
           password: this.state.passwordField.value,
         })
         .then((data) => {
-          const userData = {
-            username: data.username,
+          const userData: ILoggedInUser = {
```
Comment (oponder) [inline src/components/SignUp/SignUp.tsx:209]: I thought fixing the shape of this and calling loginSuccess would've fixed the problem, but unfortunately not. Will try to dig a bit more on Monday.

**Comment by oponder** [comment]
This should work now!

Though I made it log in for itself, instead of trying to use the auth token that passage gives it. So we can remove the login on behalf of the user part in passage now.

**Comment by oponder** [comment]
Fixes: https://github.com/giantswarm/giantswarm/issues/16698

**Review by ubergesundheit — APPROVED** [review]
LGTM

**Pull request merged**